Source/NSPasteboard.m

```diff
@@ -1939,154 +1939,182 @@ + (id) _lostServer: (NSNotification*)notification
-	    addObserver: self
-	       selector: @selector(_lostServer:)
-		   name: NSConnectionDidDieNotification
-		 object: conn];
-	}
-      else
-	{
-	  static BOOL		recursion = NO;
-	  static NSString	*cmd = nil;
-
-	  if (cmd == nil && recursion ==NO)
-	    {
-	      cmd = RETAIN([NSTask launchPathForTool: @"gpbs"]);
-	    }
-	  if (recursion == YES || cmd == nil)
-	    {
-	      NSLog(@"Unable to contact pasteboard server - "
-		@"please ensure that gpbs is running for %@.", description);
-	      return nil;
-	    }
-	  else
-	    {
-	      NSNotificationCenter *nc;
-	      NSMutableArray *startIndicator;
-	      NSArray *args = nil;
-	      NSDate *timeoutDate;
-
-	      NSDebugLLog(@"NSPasteboard",
-@"\nI couldn't contact the pasteboard server for %@ -\n"
-@"so I'm attempting to start one - which might take a few seconds.\n"
-@"Trying to launch gpbs from %@ or a machine/operating-system subdirectory.\n",
-description, [cmd stringByDeletingLastPathComponent]);
-
-	      if ([host length] > 0)
-		{
-		  args = [[NSArray alloc] initWithObjects:
-		    @"-NSHost", host,
-		    @"-GSStartupNotification", @"GSStartup-GPBS",
-		    @"--auto",
-		    nil];
-		}
-	      else
-		{
-		  args = [[NSArray alloc] initWithObjects:
-		    @"-GSStartupNotification",@"GSStartup-GPBS",
-		    @"--auto",
-		    nil];
-		}
-
-	      /*
-	      Trick: To avoid having to use global variables or new methods
-	      to track whether the notification has been received or not, we
-	      use a mutable array as an indicator. When the notification is
-	      received, the array is emptied, so we just check the count.
-	      */
-	      startIndicator = [[NSMutableArray alloc] initWithObjects:
-		AUTORELEASE([[NSObject alloc] init]), nil];
-
-	      nc = [NSDistributedNotificationCenter defaultCenter];
-	      [nc addObserver: startIndicator
-		     selector: @selector(removeAllObjects)
-			 name: @"GSStartup-GPBS"
-		       object: nil];
-
-	      [NSTask launchedTaskWithLaunchPath: cmd arguments: args];
-	      RELEASE(args);
-
-	      timeoutDate = [NSDate dateWithTimeIntervalSinceNow: 5.0];
-
-	      while ([startIndicator count]
-	        && [timeoutDate timeIntervalSinceNow] > 0.0)
-		{
-		  [[NSRunLoop currentRunLoop]
-		       runMode: NSDefaultRunLoopMode
-		    beforeDate: timeoutDate];
-		}
-
-	      [nc removeObserver: startIndicator];
-	      DESTROY(startIndicator);
-
-	      recursion = YES;
-	      [self _pbs];
-	      recursion = NO;
-	    }
-	}
+      //
+      // NOTE: This is a bit strange and bears review, I believe what TP is doing here is
+      //    using an external pasteboard server instead of gpbs.
+      //
+      // TESTPLANT-MAL-12112017: If the pasteboard server class it available
+      // the use it directly to avoid windows issue with the GPBS communications...
+      if (NSClassFromString(@"PasteboardServer") != nil)
+        {
+          the_server = [[NSClassFromString(@"PasteboardServer") alloc] init];
+        }
```

Comment:
This probably shouldn't go in? It looks like choosing to use their own class from a bundle. Perhaps they should swizzle `_lostServer:` in an injected lib.